velox/common/base/VeloxException.h

```diff
@@ -55,8 +55,18 @@ struct ExceptionContext {
   /// Calls `messageFunc(arg)` and returns the result. Returns empty string if
   /// `messageFunc` is null.
   std::string message() {
-    return messageFunc ? messageFunc(arg) : "";
+    if (!suspended) {
```

Comment:
seems unrelated to this PR?

Reply:
I'm planning to use VectorSaver in the expression handing path, hence, wanted to make sure that things will be Ok if VectorSaver itself throws. Sure enough the code went into an infinite recursion of exceptions. Hence, I made that fix. I'll extract it into a separate PR though.

Reply:
Here is a PR: https://github.com/facebookincubator/velox/pull/2578